js/evidence.js

```diff
@@ -1,26 +1,196 @@
+		}
+
+		var rect = footer.getBoundingClientRect();
+
+		return rect.bottom > 0 &&
+			rect.right > 0 &&
+			rect.left < (window.innerWidth || document.documentElement.clientWidth) &&
+			rect.top < (window.innerHeight || document.documentElement.clientHeight);
+	}
+
+	/**
+	 * Make an AJAX call and display the response.
+	 */
+	function fetch_and_display_stories(data, type) {
+		$.ajax({
+			type: 'post',
+			url: evidence.ajax_url,
+			data: data,
+			beforeSend: function () {
+				fetching = true;
+				if ('scrolled' === type) {
+					$story_container.append('<div class="loading"></div>');
+				} else if ('filtered' === type) {
+					$story_container.find('.section-wrapper').hide(400);
+				}
+			}
+		}).done(function (response) {
+			if ('scrolled' === type) {
+				$('.loading').remove();
+				$story_container.append($.parseJSON(response));
+				process_column_backgrounds();
+				$story_container.data('page', data.page + 1);
+			} else if ('filtered' === type) {
+				$story_container.html($.parseJSON(response));
+				process_column_backgrounds();
+				$story_container.data('page', 1);
+			}
+			$('.new').show(400);
+		}).always(function () {
+			fetching = false;
+		});
+	}
+
+	/**
+	 * Load more posts when the footer is reached.
+	 */
+	function infinite_scroll() {
+		window.addEventListener('scroll', function () {
+			if (fetching ||
+					all_stories ||
+					!footer_is_visible() ||
+					$story_container.data('page') === $story_container.data('total-pages') ||
+					'' !== $('#filter-options').find('option:selected').val()) {
+				return;
+			}
+
+			var data = {
+					action: 'evidence_stories',
+					nonce: evidence.nonce,
+					page: $story_container.data('page')
+				};
+
+			fetch_and_display_stories(data, 'scrolled');
+		}, 500);
+	}
+
+	/**
+	 * Scroll to the filter wrapper when an option is selected.
+	 */
+	function scroll_to_filter() {
+		var filters_top = Math.round($filters.offset().top - offset);
+
+		if ($(window).scrollTop() !== filters_top) {
+			$('html, body').animate({
+				scrollTop: filters_top
+			}, 750);
+		}
+	}
+
+	/**
+	 * Update the header text and document title when an option is selected from the filter drop-down.
+	 */
+	function update_title(name, value) {
+		var heading = value ? name : ' ',
+			doc_title = value ? name + ' | ' + evidence.default_title : evidence.default_title;
+
+		$('.topic-title').find('h3').html(heading);
+
+		document.title = doc_title;
+	}
 
-	$(document).ready( function() {
+	/**
+	 * Update the history state.
+	 */
+	function update_history(name, value) {
+		var state = { name: name, value: value },
+			url = value ? evidence.default_url + 'category/' + value + '/' : evidence.default_url;
+
+		history.pushState(state, null, url);
+	}
+
+	$(document).ready(function () {
+		var option = $(this).find('option:selected'),
+			value = option.val(),
+			name = value ? option.text() : '&nbsp;',
+			state = { name: name, value: value };
+
+		history.replaceState(state, null, null);
```

Comment:
Same as with `pushState()`, the title parameter can be `''`. I *think* we can leave off the 3rd parameter (url) if one isn't available so that it uses the current URL.